tools/topology/topology2/include/dais/alh.conf

```diff
@@ -0,0 +1,86 @@
+#
+# ALH DAI class definition. All attributes defined herein are namespaced
+# by alsatplg to "Object.Dai.ALH.N.attribute_name"
+#
+# Usage: ALH DAI objects can be instantiated as:
+#
+# Object.Dai.ALH."N" {
+# 	direction		"capture" # playback or capture
+# 	dai_name		"SDW0-Capture"
+# 	id 			2
+# 	Object.hw_config."2" {
+# 	}
+# }
+#
+# where N is the unique instance number for the ALH object within the same alsaconf node.
+
+# ALH port definition
+Class.Dai."ALH" {
+
+	#
+	# Argument used to construct DAI widget
+	#
+	# Playback DAI Index
+	DefineAttribute."dai_index" {
+		token_ref	"sof_tkn_dai.word"
+	}
+
+	DefineAttribute."direction" {
```

Comment:
Please add `token_ref` to the `direction` attribute, or the `direction` attribute will not in the final DAI structure.
`token_ref "sof_tkn_dai.string"`

Reply:
> Please add `token_ref` to the `direction` attribute, or the `direction` attribute will not in the final DAI structure.
> `token_ref "sof_tkn_dai.string"`

Thanks @aiChaoSONG According to [{SOF_TKN_DAI_DIRECTION, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32, offsetof(struct sof_ipc_comp_dai, direction), 0},](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/topology.c#L547), shouldn't it be `token_ref "sof_tkn_dai.word"`?
BTW, somehow the "direction" attribute is still not working after the change.

Reply:
> BTW, somehow the "direction" attribute is still not working after the change.

This is actually a different issue. @RanderWang has fixed it.